src/test/scala/com/mwz/sonar/scala/scapegoat/ScapegoatSensorSpec.scala

```diff
@@ -0,0 +1,367 @@
+import org.sonar.api.batch.sensor.issue.internal.DefaultIssue
+import org.sonar.api.config.internal.MapSettings
+import org.sonar.api.rule.RuleKey
+
+import java.nio.file.{Path, Paths}
+
+/** Tests the Scapegoat Sensor */
+class ScapegoatSensorSpec extends FlatSpec with SensorContextMatchers with LoneElement with OptionValues {
+  val scapegoatReportParser = new TestScapegoatReportParser()
+  val scapegoatSensor = new ScapegoatSensor(scapegoatReportParser)
+
+  behavior of "Scapegoat Sensor"
+
+  it should "correctly set descriptor" in {
+    val descriptor = new DefaultSensorDescriptor
+    scapegoatSensor.describe(descriptor)
+
+    descriptor should not be 'global
+    descriptor.name shouldBe "Scapegoat Sensor"
+    descriptor.`type` shouldBe InputFile.Type.MAIN
+    descriptor.languages.loneElement shouldBe "scala"
+    descriptor.ruleRepositories.loneElement shouldBe "sonar-scala-scapegoat-repository"
+  }
+
+  it should "get default scapegoat report path, when the scala version property is missing" in {
+    val reportPath = scapegoatSensor.getScapegoatReportPath(new MapSettings().asConfig())
+
+    reportPath shouldBe Paths.get("target", "scala-2.11", "scapegoat-report", "scapegoat.xml")
+  }
+
+  it should "get default scapegoat report path, when the scala version property is set" in {
+    val reportPath = scapegoatSensor.getScapegoatReportPath(
+      new MapSettings().setProperty("sonar.scala.version", "2.12.6").asConfig()
+    )
+
+    reportPath shouldBe Paths.get("target", "scala-2.12", "scapegoat-report", "scapegoat.xml")
+  }
+
+  it should "get scapegoat report path set in sonar properties" in {
+    val reportPath = scapegoatSensor.getScapegoatReportPath(
+      new MapSettings().setProperty("sonar.scala.scapegoat.reportPath", "target/report-path").asConfig()
+    )
+
+    reportPath shouldBe Paths.get("target", "report-path")
+  }
+
+  it should "get a module file using the filename extracted from the scapegoat report" in {
+    val filesystem = new DefaultFileSystem(Paths.get("./"))
+    val testFileA = TestInputFileBuilder
+      .create("", "src/main/scala/com/mwz/sonar/scala/scapegoat/TestFileA.scala")
+      .setLanguage("scala")
+      .setType(InputFile.Type.MAIN)
+      .build()
+    filesystem.add(testFileA)
+    val moduleFile =
+      scapegoatSensor.getModuleFile("com/mwz/sonar/scala/scapegoat/TestFileA.scala", filesystem)
+
+    moduleFile.value shouldBe testFileA
+  }
+
+  it should "not get a module file if its filename does not match any file" in {
+    val filesystem = new DefaultFileSystem(Paths.get("./"))
+    val moduleFile =
+      scapegoatSensor.getModuleFile("com/mwz/sonar/scala/scapegoat/TestFileA.scala", filesystem)
+
+    moduleFile shouldBe None
+  }
+
+  it should "create an issue for each scapegoat report's warning" in {
+    // create the sensor context
+    val sensorContext = SensorContextTester.create(Paths.get("./"))
```

Comment:
Perhaps you could use `cwd` from `PathUtils` here?

Reply:
Well if you really want to.
But here I only need to instantiate the sensor, the module base path doesn't affect the test in any way, since all paths are relative.